wgsl/index.bs

```diff
@@ -565,37 +567,168 @@ as the first parameter.
-    There are up to four channels, named `r`, `g`, `b`, or `a`.
+    A texel format has up to four channels: `r`, `g`, `b`, and `a`,
+    normally corresponding to the concepts of red, green, blue, and alpha channels.
 : <dfn noexport>channel format</dfn>
 :: The number of bits in the channel, and how those bits are interpreted.
-    In some cases different channels in a texel will have different channel types.
+
+Only certain texel formats are used in [SHORTNAME] source code.
+The channel formats used to define those texel formats are listed in the following table.
+The last column specfies the conversion from the stored channel bits to the value used in the shader.
+This is also known as the <dfn noexport>channel transfer function</dfn>, or CTF.
 
 <table class='data'>
-  <caption>Channel formats</caption>
+  <caption>Channel Formats</caption>
   <thead>
-    <tr><td>Channel format<td>Number of stored bits<td>Interpetation of stored bits<td>Shader type<td>Shader value
+    <tr><td>Channel format<td>Number of stored bits<td>Interpetation of stored bits<td>Shader type<td width="25%">Shader value
+(Channel Transfer Function)
   </thead>
   <tr><td>8unorm<td>8<td>unsigned integer |v| &isinv; {0,...,255}<td>f32<td> |v| &div; 255
-  <tr><td>8snorm<td>8<td>signed integer |v| &isinv; {-128,...,127}<td>f32<td> |v| &div; 128
+  <tr><td>8snorm<td>8<td>signed integer |v| &isinv; {-128,...,127}<td>f32<td> max(-1, |v| &div; 127)
   <tr><td>8uint<td>8<td>unsigned integer |v| &isinv; {0,...,255}<td>u32<td> |v| &div; 255
-  <tr><td>8sint<td>8<td>signed integer |v| &isinv; {-128,...,127}<td>i32<td> |v| &div; 128
-  <tr><td>16uint<td>16<td>unsigned integer |v| &isinv; {0,...,65535}<td>u32<td> |v| &div; 65535
-  <tr><td>16sint<td>16<td>signed integer |v| &isinv; {-32768,...,32767}<td>i32<td> |v| &div; 32768
-  <tr><td>16float<td>16<td>IEEE 754 16-bit floating point value |v|, with 1 sign bit, 5 exponent bits, 10 mantissa bits<td>f32<td>|v|, converted to f32
+  <tr><td>8sint<td>8<td>signed integer |v| &isinv; {-128,...,127}<td>i32<td> max(-1, |v| &div; 127)
+  <tr><td>16uint<td>16<td>unsigned integer |v| &isinv; {0,...,65535}<td>u32<td> |v|
+  <tr><td>16sint<td>16<td>signed integer |v| &isinv; {-32768,...,32767}<td>i32<td> |v|
+  <tr><td>16float<td>16<td>IEEE 754 16-bit floating point value |v|, with 1 sign bit, 5 exponent bits, 10 mantissa bits<td>f32<td>|v|
   <tr><td>32uint<td>32<td>32-bit signed integer value |v|<td>u32<td>|v|
   <tr><td>32sint<td>32<td>32-bit unsigned integer value |v|<td>i32<td>|v|
   <tr><td>32float<td>32<td>IEEE 754 32-bit floating point value |v|<td>f32<td>|v|
 </table>
 
-TODO: check the formulae
+The texel formats listed in the following tables are used to parameterize storage texture types.
+See [[#texture-ro]] and [[#texture-wo]].
+The last column uses the format-specific [=channel transfer function=] from the channel formats table.
+
+When the texel format does not have a green, blue, or alpha channel, the
```

Comment:
We can't have that in the language. Is it truly undefined? Could you make an issue/mini-investigation about this?

Reply:
I filed #1115 for this.

Reply:
Let's have an `Issue(gpuweb/1115): xxx` for it here (or whatever the syntax is)

Reply:
After that discussion evolved, I've made missing green and blue map to 0, and alpha to 1.